pilot/pkg/config/kube/gateway/inferencepool_collection.go

```diff
@@ -0,0 +1,319 @@
+	pools krt.Collection[*inferencev1alpha2.InferencePool],
+	services krt.Collection[*corev1.Service],
+	httpRoutes krt.Collection[*gateway.HTTPRoute],
+	gateways krt.Collection[*gateway.Gateway],
+	routesByNamespace krt.Index[string, *gateway.HTTPRoute],
+	c *Controller,
+	opts krt.OptionsBuilder,
+) (krt.StatusCollection[*inferencev1alpha2.InferencePool, inferencev1alpha2.InferencePoolStatus], krt.Collection[InferencePool]) {
+	return krt.NewStatusCollection(pools,
+		func(
+			ctx krt.HandlerContext,
+			pool *inferencev1alpha2.InferencePool,
+		) (*inferencev1alpha2.InferencePoolStatus, *InferencePool) {
+			// First, let's build the shadow service
+			extRef := extRefInfo{
+				name: string(pool.Spec.ExtensionRef.Name),
+			}
+			if pool.Spec.ExtensionRef.PortNumber != nil {
+				extRef.port = int32(*pool.Spec.ExtensionRef.PortNumber)
+			} else {
+				extRef.port = 9002 // Default port for the inference extension
+			}
+
+			svcName, err := InferencePoolServiceName(pool.Name)
+			if err != nil {
+				log.Errorf("failed to generate service name for InferencePool %s: %v", pool.Name, err)
+				return nil, nil
+			}
+
+			shadowSvcInfo := shadowServiceInfo{
+				key: types.NamespacedName{
+					Name:      svcName,
+					Namespace: pool.GetNamespace(),
+				},
+				selector:   make(map[string]string, len(pool.Spec.Selector)),
+				poolName:   pool.GetName(),
+				targetPort: pool.Spec.TargetPortNumber,
+				poolUID:    pool.GetUID(),
+			}
+
+			for k, v := range pool.Spec.Selector {
+				shadowSvcInfo.selector[string(k)] = string(v)
+			}
+
+			// Make sure we reconcile the service if it is manually changed
+			_ = krt.FetchOne(ctx, services, krt.FilterKey(shadowSvcInfo.key.String()))
+
+			gatewayParentsToEnsure := sets.New[types.NamespacedName]()
+			routeList := krt.Fetch(ctx, httpRoutes, krt.FilterIndex(routesByNamespace, pool.Namespace))
+			for _, r := range routeList {
+				for _, rule := range r.Spec.Rules {
+					for _, httpBackendRef := range rule.BackendRefs {
+						if httpBackendRef.BackendRef.Group == nil || httpBackendRef.BackendRef.Kind == nil {
+							continue
+						}
+						if string(*httpBackendRef.BackendRef.Group) == gvk.InferencePool.Group &&
+							string(*httpBackendRef.BackendRef.Kind) == gvk.InferencePool.Kind &&
+							string(httpBackendRef.BackendRef.Name) == pool.ObjectMeta.Name {
+							for _, p := range r.Status.Parents {
+								if supportedControllers.Contains(p.ControllerName) {
+									ns := r.Namespace
+									if p.ParentRef.Namespace != nil && *p.ParentRef.Namespace != "" {
+										ns = string(*p.ParentRef.Namespace)
+									}
+									gatewayParentsToEnsure.Insert(types.NamespacedName{Name: string(p.ParentRef.Name), Namespace: ns})
+								}
+							}
+						}
+					}
+				}
+			}
+
+			newParents := []inferencev1alpha2.PoolStatus{}
+			for gtw := range gatewayParentsToEnsure {
+				newParents = append(newParents, *poolStatusTmpl(gtw.Name, gtw.Namespace, pool.Generation))
```

Comment:
This is going to wipe out any other controllers. We need the same logic as HTTPRoute has

Reply:
Added